python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/utils.py

```diff
@@ -267,7 +267,30 @@ def _update_common_interface_cfg(
             interface.ip_address = p2p_link_data["ip"]
 
         if p2p_link.include_in_underlay_protocol:
-            if p2p_link.underlay_multicast and self.shared_utils.underlay_multicast:
+            multicast_pim_sm = default(p2p_link.multicast_pim_sm, self.shared_utils.underlay_multicast_pim_sm_enabled)
+            if multicast_pim_sm and self.shared_utils.underlay_multicast_pim_sm_enabled:
+                interface.pim.ipv4.sparse_mode = True
+            elif multicast_pim_sm and not self.shared_utils.underlay_multicast_pim_sm_enabled:
+                msg = (
+                    f"{self.data_model}.p2p_links has `include_in_underlay_protocol: true` and "
+                    "`multicast_pim_sm: true`, which requires the global setting `underlay_multicast_pim_sm` "
+                    "to also be set to `true`."
+                )
+
+                raise AristaAvdInvalidInputsError(msg)
```

Comment:
```suggestion
             if p2p_link.multicast_pim_sm and not self.shared_utils.underlay_multicast_pim_sm_enabled:
                msg = (
                    f"{self.data_model}.p2p_links has `include_in_underlay_protocol: true` and "
                    "`multicast_pim_sm: true`, which requires the global setting `underlay_multicast_pim_sm` "
                     "to also be set to `true`."
                )
                raise AristaAvdInvalidInputsError(msg)
             elif p2p_link.multicast_pim_sm or self.shared_utils.underlay_multicast_pim_sm_enabled: 
                interface.pim.ipv4.sparse_mode = True
```

Reply:
By this if we set the p2p_link.multicast_pim_sm: false and global underlay_multicast_pim_sm: true it still render the pim sm config which is wrong. If globally enabled and we are disabling in a specific p2p link then this logic wont work.

Reply:
we still need to simplify the logic with default it is very hard to understand as we are defaulting to a vlaue we are comparing ourselves against after which makes feel like this is nonsensical

```suggestion
             if p2p_link.multicast_pim_sm
                if not self.shared_utils.underlay_multicast_pim_sm_enabled:
                    msg = (
                        f"{self.data_model}.p2p_links has `include_in_underlay_protocol: true` and "
                        "`multicast_pim_sm: true`, which requires the global setting `underlay_multicast_pim_sm` "
                         "to also be set to `true`."
                    )
                    raise AristaAvdInvalidInputsError(msg)
                interface.pim.ipv4.sparse_mode = True
             elif p2p_link.multicast_pim_sm is not False and  self.shared_utils.underlay_multicast_pim_sm_enabled: 
                interface.pim.ipv4.sparse_mode = True
```

Reply:
yes this will work, updated the same